common/src/main/java/com/wynntils/models/map/pois/SeaskipperPoi.java

```diff
@@ -0,0 +1,219 @@
+        }
+
+        BufferedRenderUtils.drawRect(
+                poseStack,
+                bufferSource,
+                color.withAlpha(65),
+                actualRenderX,
+                actualRenderZ,
+                0,
+                renderWidth,
+                renderHeight);
+
+        BufferedRenderUtils.drawRectBorders(
+                poseStack,
+                bufferSource,
+                color,
+                actualRenderX,
+                actualRenderZ,
+                actualRenderX + renderWidth,
+                actualRenderZ + renderHeight,
+                0,
+                1.5f);
+
+        BufferedFontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        bufferSource,
+                        destination,
+                        actualRenderX,
+                        actualRenderX + renderWidth,
+                        actualRenderZ,
+                        actualRenderZ + renderHeight,
+                        0,
+                        color,
+                        HorizontalAlignment.CENTER,
+                        VerticalAlignment.TOP,
+                        TextShadow.OUTLINE);
+
+        poseStack.popPose();
+    }
+
+    public void renderAtWithoutBorders(
+            PoseStack poseStack, MultiBufferSource bufferSource, float renderX, float renderY, float mapZoom) {
+        poseStack.pushPose();
+        poseStack.translate(0, 0, 100);
+
+        final float renderWidth = width * mapZoom;
+        final float renderHeight = height * mapZoom;
+        final float actualRenderX = renderX - renderWidth / 2f;
+        final float actualRenderZ = renderY - renderHeight / 2f;
+
+        CustomColor color;
+
+        if (isPlayerAtPoi) {
+            color = CommonColors.ORANGE;
+        } else {
+            color = CommonColors.WHITE;
+        }
+
+        BufferedFontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        bufferSource,
+                        destination,
+                        actualRenderX,
+                        actualRenderX + renderWidth,
+                        actualRenderZ,
+                        actualRenderZ + renderHeight,
+                        0,
+                        color,
+                        HorizontalAlignment.CENTER,
+                        VerticalAlignment.TOP,
+                        TextShadow.OUTLINE);
+
+        poseStack.popPose();
+    }
+
+    @Override
+    public int getWidth(float mapZoom, float scale) {
+        return (int) (width * mapZoom);
+    }
+
+    @Override
+    public int getHeight(float mapZoom, float scale) {
+        return (int) (width * mapZoom);
+    }
+
+    @Override
+    public String getName() {
+        return destination;
+    }
+
+    public int getLevel() {
+        return combatLevel;
+    }
+
+    public boolean isPlayerInside(Vec3 playerPosition) {
```

Comment:
I think you should figure this out in the POI constructor. Then the field can be final, and this can be a simple getter.